src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -1155,26 +1191,41 @@ private void ReduceConcatenateWithAutoAtomic()
 
                 static void ProcessNode(RegexNode node, RegexNode subsequent, int maxDepth)
                 {
-                    // Skip down the node past irrelevant nodes.  We don't need to
-                    // skip Groups, as they should have already been reduced away.
-                    // If there's a concatenation, we can jump to the last element of it.
-                    while (node.Type == Capture || node.Type == Concatenate)
+                    // Skip down the node past irrelevant nodes.
+                    while (true)
                     {
-                        node = node.Child(node.ChildCount() - 1);
+                        // We can always recur into captures and into the last node of concatenations.
```

Comment:
```suggestion
                        // We can always recurse into captures and into the last node of concatenations.
```

Reply:
Well, maybe "recur" is a possible word to use, but "recurse" is more common. Either is fine I guess.

Reply:
I had it drilled into me in college that "recur" is what happens when you employ recursion, and "recourse" is what happens when you swear again. Ever since "recurse" just sounds wrong to me.